x-pack/plugins/security/server/authentication/index.ts

```diff
@@ -30,35 +31,32 @@ export {
 } from './api_keys';
 
 interface SetupAuthenticationParams {
-  core: CoreSetup;
+  http: CoreSetup['http'];
   clusterClient: IClusterClient;
   config: ConfigType;
+  license: SecurityLicense;
   loggers: LoggerFactory;
-  getLegacyAPI(): LegacyAPI;
+  getLegacyAPI(): Pick<LegacyAPI, 'isSystemAPIRequest'>;
```

Comment:
__note:__ trying to specify/pick only used properties of the legacy API to make it more auditable.